compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -924,7 +924,8 @@ protected Result seekMatch(ParserRuleContext currentCtx, int lookahead, int curr
                     hasMatch = nextToken.kind == SyntaxKind.CLOSE_PAREN_TOKEN;
                     break;
                 case OBJECT_CONSTRUCTOR_QUALIFIER:
-                    hasMatch = nextToken.kind == SyntaxKind.CLIENT_KEYWORD;
+                    hasMatch = nextToken.kind == SyntaxKind.CLIENT_KEYWORD ||
+                            nextToken.kind == SyntaxKind.ISOLATED_KEYWORD;
```

Comment:
Do we need to address `isolated client object {..` case in error handler?

Reply:
Currently, it is not handled right?

Reply:
In error handler we don't have a proper way to solve this issue. But, in a very few cases we have handled it like below.
```java
    private static final ParserRuleContext[] OBJECT_CONSTRUCTOR_START =
            { ParserRuleContext.OBJECT_KEYWORD, ParserRuleContext.ISOLATED_KEYWORD, 
              ParserRuleContext.CLIENT_KEYWORD };

    private static final ParserRuleContext[] OBJECT_CONSTRUCTOR_WITHOUT_ISOLATED =
            { ParserRuleContext.OBJECT_KEYWORD, ParserRuleContext.CLIENT_KEYWORD };

    private static final ParserRuleContext[] OBJECT_CONSTRUCTOR_WITHOUT_CLIENT =
            { ParserRuleContext.OBJECT_KEYWORD, ParserRuleContext.ISOLATED_KEYWORD };
```

But, my concern with above logic is; it adds **cyclic** alternative paths.
For example, something like `isolated client isolated client isolated` is an alternative path.

So, If we assume `LOOKAHEAD_LIMIT = 5` and entry point is `OBJECT_CONSTRUCTOR_START`, 
In 2 qualifier case: we check for `2 + 2x1 + 2x1x1 + 2x1x1x1 + 2x1x1x1x1 = 10` alternative qualifier paths.
In 4 qualifier case: we check for `4 + 4x3 + 4x3x3 + 4x3x3x3 + 4x3x3x3x3 = 484` alternative qualifier paths.
which are all unnecessary. (IINM)

We could rather define rules like below. (aliasing `OBJECT_CONSTRUCTOR_QUALIFIER`)
```java
case FIRST_OBJECT_CONSTRUCTOR_QUALIFIER:
case SECOND_OBJECT_CONSTRUCTOR_QUALIFIER:
       hasMatch = nextToken.kind == SyntaxKind.CLIENT_KEYWORD || 
                  nextToken.kind == SyntaxKind.ISOLATED_KEYWORD;
       break;
```
In 2 qualifier case: we check for 2 alternative qualifier paths.
In 4 qualifier case: we check for 4 alternative qualifier paths.

or do we need to worry about this? @SupunS @rdulmina

Reply:
I think second option is a viable and a simple solution.

Reply:
+1